File: javascript/matrix/matrix.js
export class Matrix {
  constructor(matrix_string) {
    this.data = [];
    for (const row of matrix_string.split("\n")) {
      this.data.push(row.split(" ").map(Number));
    }
  }

  get rows() {
    return this.data;
  }

  get columns() {
    let columns = [];
    for (let i = 0; i < this.data.length; i++) {
      columns.push(this.data.map(function(row) {return row[i];}))
    }
    return columns;
  }
}

File: python/matrix/matrix.py
class Matrix(object):
    def __init__(self, matrix_string):
        self.data = []
        for row in matrix_string.split("\n"):
            self.data.append(map(int, row.split(" ")))

    def row(self, index):
        return self.data[index - 1]

    def column(self, index):
        return [row[index - 1] for row in self.data]
